src/core/experimental/memutils.d

```diff
@@ -0,0 +1,332 @@
+/**
+ * Pure D replacement of the C Standard Library basic memory building blocks of string.h
+ * Source: $(DRUNTIMESRC core/experimental/memutils.d)
+ */
+module core.experimental.memutils;
+
+unittest
+{
+    Dmemset_testStaticType!(byte)(5);
+    Dmemset_testStaticType!(ubyte)(5);
+    Dmemset_testStaticType!(short)(5);
+    Dmemset_testStaticType!(ushort)(5);
+    Dmemset_testStaticType!(int)(5);
+    Dmemset_testStaticType!(uint)(5);
+    Dmemset_testStaticType!(long)(5);
+    Dmemset_testStaticType!(ulong)(5);
+    Dmemset_testStaticType!(float)(5);
+    Dmemset_testStaticType!(double)(5);
+    Dmemset_testStaticType!(real)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 3);
+    static foreach (i; 1..10) {
+        Dmemset_testDynamicArray!(ubyte)(5, 2^^i);
+        Dmemset_testStaticArray!(ubyte, 2^^i)(5);
+    }
+    Dmemset_testDynamicArray!(ubyte)(5, 100);
+    Dmemset_testStaticArray!(ubyte, 100)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 500);
+    Dmemset_testStaticArray!(ubyte, 500)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 700);
+    Dmemset_testStaticArray!(ubyte, 700)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 3434);
+    Dmemset_testStaticArray!(ubyte, 3434)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 7128);
+    Dmemset_testStaticArray!(ubyte, 7128)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 13908);
+    Dmemset_testStaticArray!(ubyte, 13908)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 16343);
+    Dmemset_testStaticArray!(ubyte, 16343)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 27897);
+    Dmemset_testStaticArray!(ubyte, 27897)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 32344);
+    Dmemset_testStaticArray!(ubyte, 32344)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 46830);
+    Dmemset_testStaticArray!(ubyte, 46830)(5);
+    Dmemset_testDynamicArray!(ubyte)(5, 64349);
+    Dmemset_testStaticArray!(ubyte, 64349)(5);
+}
+
+// From a very good Chandler Carruth video on benchmarking: https://www.youtube.com/watch?v=nXaxk27zwlk
+void escape(void* p)
+{
+    version (LDC)
+    {
+        import ldc.llvmasm;
+        __asm("", "r,~{memory}", p);
+    }
+    version (GNU)
+    {
+        asm { "" : : "g" p : "memory"; }
+    }
+}
+
+void Dmemset_verifyArray(T)(int j, const ref T[] a, const ubyte v)
+{
```

Comment:
Also, if as this is for testing purposes only it should be `private` and probably even have the `CoreUnittest` version around it to make sure it doesn't end up in the released sources. If you don't like the version tags, you could create a folder in `tests` and move all the test functions over there.

Reply:
Done, but not sure if done correctly.